### PR TITLE
prowgen: use one all-powerful pull secret

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

Tested here: https://github.com/openshift/release/pull/14858

/assign @hongkailiu @petr-muller

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Part of DPTP-1845

File: cmd/ci-operator-prowgen/testdata/zz_fixture_postsubmit_TestFromCIOperatorConfigToProwYaml_Using_a_variant_config__one_test_and_images__one_existing_job._Expect_one_presubmit__pre_post_submit_images_jobs._Existing_job_should_not_be_changed..yaml [hunk 1]
```diff
@@ -70,7 +70,7 @@ postsubmits:
       volumes:
       - name: pull-secret
         secret:
-          secretName: registry-pull-credentials
+          secretName: registry-pull-credentials-all
```
Comment (petr-muller) [inline cmd/ci-operator-prowgen/testdata/zz_fixture_postsubmit_TestFromCIOperatorConfigToProwYaml_Using_a_variant_config__one_test_and_images__one_existing_job._Expect_one_presubmit__pre_post_submit_images_jobs._Existing_job_should_not_be_changed..yaml:73]: So when we're doing this, maybe we can also make the volume name and the path consistent with the secret name to make the reasoning about jobs simpler?

File: pkg/prowgen/prowgen.go [hunk 2]
```diff
@@ -263,21 +263,9 @@ func generateCiOperatorPodSpec(info *ProwgenInfo, secrets []*cioperatorapi.Secre
...
-		// we need to create a pull secret in the namespace that ci-operator
-		// runs in. While the --secret-dir mechanism is *meant* to provide
-		// secrets to the tests themselves, this secret will have no consumer
-		// and that is OK. We just need it to exist in the test namespace so
-		// that the image import controller can use it.
-		secrets = append(secrets, &cioperatorapi.Secret{
-			Name: "ci-pull-credentials",
-		})
-	}
-	podSpec := generateCiOperatorPodSpec(info, secrets, []string{test.As})
```
Comment (petr-muller) [inline pkg/prowgen/prowgen.go:280]: This removal is a little surprising to me but I assume you know what you're doing and that the ` --image-import-pull-secret` is sufficient for everything that `ci-operator` does.

File: cmd/ci-operator-prowgen/testdata/zz_fixture_postsubmit_TestFromCIOperatorConfigToProwYaml_Using_a_variant_config__one_test_and_images__one_existing_job._Expect_one_presubmit__pre_post_submit_images_jobs._Existing_job_should_not_be_changed..yaml (hunk 1 above)
Comment (stevekuznetsov) [reply cmd/ci-operator-prowgen/testdata/zz_fixture_postsubmit_TestFromCIOperatorConfigToProwYaml_Using_a_variant_config__one_test_and_images__one_existing_job._Expect_one_presubmit__pre_post_submit_images_jobs._Existing_job_should_not_be_changed..yaml:73]: Can we get the functional bit in, then rename? I want to get rid of the old secret and have `registry-pull-credentials` everywhere

File: pkg/prowgen/prowgen.go (hunk 2 above)
Comment (stevekuznetsov) [reply pkg/prowgen/prowgen.go:280]: Yeah, I tested it - that flag will both create the secret in the namespace *and* use it's content for internal stuff

File: cmd/ci-operator-prowgen/testdata/zz_fixture_postsubmit_TestFromCIOperatorConfigToProwYaml_Using_a_variant_config__one_test_and_images__one_existing_job._Expect_one_presubmit__pre_post_submit_images_jobs._Existing_job_should_not_be_changed..yaml (hunk 1 above)
Comment (petr-muller) [reply cmd/ci-operator-prowgen/testdata/zz_fixture_postsubmit_TestFromCIOperatorConfigToProwYaml_Using_a_variant_config__one_test_and_images__one_existing_job._Expect_one_presubmit__pre_post_submit_images_jobs._Existing_job_should_not_be_changed..yaml:73]: /shrug

**Review by petr-muller — APPROVED** [review]
/hold

Need matching o/r PR

**Comment by stevekuznetsov** [comment]
I'll write a PR for release

**Comment by stevekuznetsov** [comment]
/hold
sigh how did this rot so quickly

**Pull request closed without merging**